graph_snyk_cve_sync.py

```diff
@@ -100,55 +103,86 @@ def force_run_ingestion(self):
         """Return if ingestion mode is on."""
         return os.environ.get('SNYK_INGESTION_FORCE_RUN', 'false').lower() in ('1', 'yes', 'true')
 
+    def is_delta_mode_on(self):
+        """Return if the delta feed mode is on."""
+        return os.environ.get('SNYK_DELTA_FEED_MODE', 'false').lower() in ('1', 'yes', 'true')
+
 
 class SnykCveSync:
     """Snyk class to sync cves to graph."""
 
-    def __init__(self):
+    def __init__(self, data):
         """Init method for helper class."""
         self.helper = Helper()
         self.CVE_DATA = {}
-
-    def get_version_rules_npm(self, vuln_versions):
-        """Version rules for npm eco."""
-        rules = []
-        for version in vuln_versions:
-            rules.append(NpmSpec(version))
-        return rules
+        self.utc_now = datetime.utcnow()
+        if not data:
+            self.snyk_data = self.helper.read_data_from_s3(self.utc_now.strftime('%d-%m-%y'),
+                                                           "snyk-feed/")
+        else:
+            self.snyk_data = data
+
+    def parse_data_for_eco(self, eco):
+        """Return True/False depending on whether the ecosysytem data should be parsed or not."""
+        if self.helper.is_delta_mode_on():
+            return True
+        day = str(self.utc_now.weekday())
```

Comment:
This method is getting called the number of times the loop runs. It can be taken out and passed as a parameter.

Reply:
noted

Reply:
done